### PR TITLE
Feature/maboss updates

### PR DESCRIPTION
**Pull request by tjsego**
This PR is downstream of https://github.com/CompuCell3D/CompuCell3DExtensions/pull/2

* Integrated cell inventory events and MaBoSS container for major speedup in MaBoSS-integrated simulations
* Added convenience features for getting model population statistics, setting initial states, etc.

### CODE REVIEW

File: cc3d/core/MaBoSSCC3D.py [hunk 1]
```diff
@@ -16,30 +16,48 @@
...
     pass
 
 
+class MaBoSSInventoryWatcher(CellInventoryWatcher):
+    """
+    Inventory watcher that maintains synchronization between the cell and engine inventories
+    """
+
+    def on_cell_remove(self, cell: CompuCell.CellG):
+        from cc3d.CompuCellSetup import persistent_globals as pg
```
Comment (CompuCell3D) [inline cc3d/core/MaBoSSCC3D.py:44]: can we move this import to the top of the file?

File: cc3d/core/MaBoSSCC3D.py [hunk 2]
```diff
@@ -146,9 +201,20 @@ def add_maboss_to_cell(cell: CompuCell.CellG,
                               time_step=time_step,
                               time_tick=time_tick,
                               discrete_time=discrete_time,
-                              seed=seed)
+                              seed=seed,
+                              istate=istate)
         if CompuCell.CellG.__maboss__ not in cell.dict.keys():
             cell.dict[CompuCell.CellG.__maboss__] = {}
+
+        from cc3d.CompuCellSetup import persistent_globals as pg
```
Comment (CompuCell3D) [inline cc3d/core/MaBoSSCC3D.py:209]: move import to the top of the file?

File: cc3d/core/MaBoSSCC3D.py [hunk 3]
```diff
@@ -202,22 +274,34 @@ def maboss_model(bnd_file: str = None,
...
         """
-        maboss_key = CompuCell.CellG.__maboss__
-        for cell in self.cell_list:
-            try:
-                maboss_models: Dict[str, MaBoSSCC3DPy.CC3DMaBoSSEngine] = cell.dict[maboss_key]
-                for model in maboss_models.values():
-                    model.step()
-            except KeyError:
-                pass
+        from cc3d.CompuCellSetup import persistent_globals as pg
```
Comment (CompuCell3D) [inline cc3d/core/MaBoSSCC3D.py:286]: top of the file?

File: cc3d/core/MaBoSSCC3D.py [hunk 4]
```diff
@@ -202,22 +274,34 @@ def maboss_model(bnd_file: str = None,
...
+        Returns a dictionary with summary statistics of a node of a model, including
+
+        - ``num_true``: the total number of cells with the model and a true node state
+        - ``num_false``: the total number of cells with the model and a false node state
+
+        :param model_name: name of model
+        :param node_name: name of node
+        :return: dictionary of statistics
+        """
+        from cc3d.CompuCellSetup import persistent_globals as pg
```
Comment (CompuCell3D) [inline cc3d/core/MaBoSSCC3D.py:301]: ditto

**Review by CompuCell3D — CHANGES_REQUESTED** [review]
Only minor comments. Otherwise it looks very good,

**Comment by tjsego** [comment]
As a reminder, this will break without the following: https://github.com/CompuCell3D/CompuCell3DExtensions/pull/2

File: cc3d/core/MaBoSSCC3D.py (hunk 1 above)
Comment (tjsego) [reply cc3d/core/MaBoSSCC3D.py]: Done

File: cc3d/core/MaBoSSCC3D.py (hunk 2 above)
Comment (tjsego) [reply cc3d/core/MaBoSSCC3D.py]: Done

File: cc3d/core/MaBoSSCC3D.py (hunk 3 above)
Comment (tjsego) [reply cc3d/core/MaBoSSCC3D.py]: Done

File: cc3d/core/MaBoSSCC3D.py (hunk 4 above)
Comment (tjsego) [reply cc3d/core/MaBoSSCC3D.py]: Done

**Pull request merged**